### PR TITLE
GoogleUtilities unit tests on iOS 8

### LINKED ISSUE #3091: GoogleUtilities unit tests should run for iOS 8 targets
Screen Shot 2019-05-27 at 9 09 41 AM

### PR DESCRIPTION
**Pull request by maksymmalyhin**
Fixes #3091.

### CODE REVIEW

File: GoogleUtilities/SwizzlerTestHelpers/GULRuntimeClassSnapshot.m [hunk 1]
```diff
@@ -132,10 +132,9 @@ - (void)captureSelectorsAndImps {
     IMP imp = method_getImplementation(method);
     NSString *impString =
         [NSString stringWithFormat:@"%p -[%@ %@]", imp, NSStringFromClass(_aClass), methodString];
-    NSAssert(![_imps containsObject:impString],
```
Comment (paulb777) [inline GoogleUtilities/SwizzlerTestHelpers/GULRuntimeClassSnapshot.m]: Why is it ok to remove this assert?

File: GoogleUtilities/Example/Tests/Swizzler/GULAppDelegateSwizzlerTest.m [hunk 2]
```diff
@@ -667,72 +674,76 @@ - (void)testAppDelegateInstance {
...
-  [testAppDelegate application:[GULApplication sharedApplication]
-                       openURL:testURL
-                       options:testOpenURLOptions];
-  OCMVerifyAll(interceptor);
-  OCMVerifyAll(interceptor2);
-
-  // Check that original implementation was called with proper parameters
-  XCTAssertEqual(testAppDelegate.application, [GULApplication sharedApplication]);
-  XCTAssertEqual(testAppDelegate.url, testURL);
+  if (@available(iOS 10, *)) {
```
Comment (paulb777) [inline GoogleUtilities/Example/Tests/Swizzler/GULAppDelegateSwizzlerTest.m]: Should there be alternate tests for iOS 9 and 10?

File: GoogleUtilities/SwizzlerTestHelpers/GULRuntimeClassSnapshot.m (hunk 1 above)
Comment (maksymmalyhin) [reply GoogleUtilities/SwizzlerTestHelpers/GULRuntimeClassSnapshot.m]: @paulb777 Good question! It fails on `-[SBSStatusBarStyleOverridesAssertionData isExclusive]`. It looks like `isExclusive` is defined like both a property getter and a method ([the following header](https://github.com/nst/iOS-Runtime-Headers/blob/master/PrivateFrameworks/SpringBoardServices.framework/SBSStatusBarStyleOverridesAssertionData.h) may be a proof of it) that's why, I guess, it is detected as a duplicate. Based on it, just a condition should be fine. I could add a log message here as well, but we don't have a proper logger available in the class and I would not like to add it for the Release configuration.
WDYT?

File: GoogleUtilities/Example/Tests/Swizzler/GULAppDelegateSwizzlerTest.m (hunk 2 above)
Comment (maksymmalyhin) [reply GoogleUtilities/Example/Tests/Swizzler/GULAppDelegateSwizzlerTest.m]: Sure, there is one, see tests around `[UIApplication application:openURL:sourceApplication:annotation:]`

File: GoogleUtilities/SwizzlerTestHelpers/GULRuntimeClassSnapshot.m (hunk 1 above)
Comment (paulb777) [reply GoogleUtilities/SwizzlerTestHelpers/GULRuntimeClassSnapshot.m]: SGTM

**Pull request merged**